test/metabase/sync_database/analyze_test.clj

```diff
@@ -72,3 +79,78 @@
 (expect false (values-are-valid-emails? [100]))
 (expect false (values-are-valid-emails? ["true"]))
 (expect false (values-are-valid-emails? ["false"]))
+
+;; Tests to avoid analyzing hidden tables
+(defn count-unanalyzed-fields [db_id table-name]
+  (let [table-id (db/select-one-id Table, :db_id db_id, :active true :name table-name)]
+    (assert (pos? ;; don't let ourselves be fooled if the test passes because the table is
+                  ;; totally broken or has no fields. Make sure we actually test something
+             (db/count Field :table_id table-id)))
+    (db/count Field :last_analyzed nil :table_id table-id)))
+
+(defn get-latest-sync-time [db_id table-name]
```

Comment:
since everywhere you call this you already have the full table why not just make this function take `[table]`.

Also I think we ought to stick to the adage 

> Pure function names should be nouns describing the value they return.

and call this something like `latest-sync-time` (drop the `get-`).

See https://stuartsierra.com/2016/01/09/how-to-name-clojure-functions and https://github.com/metabase/metabase/wiki/Metabase-Clojure-Style-Guide#pure-function-names-should-be-nouns